### PR TITLE
Fix DB sessions timeouts

### PR DESCRIPTION
**Pull request by sbrunner**
Fix: https://sentry.io/organizations/camptocamp/issues/1452169136/?environment=prod-2-5&project=1851011&query=is%3Aunresolved, https://sentry.io/organizations/camptocamp/issues/1384751595/?environment=prod-2-5&project=1851011&query=is%3Aunresolved

### CODE REVIEW

**Comment by arnaud-morvan** [comment]
Could we have a description about what's the problem and the resolution.
I suppose it is related to session persistence between requests.
Maybe having a fresh session for each request could be enought.

File: docker/qgisserver/geomapfish_qgisserver/accesscontrol.py [hunk 1]
```diff
@@ -277,7 +277,10 @@ def browse(path, node):
             # Transform ancestor names in LayerWMS instances
             layers = {}  # type: Dict[str, List[LayerWMS]]
             for layer in (
-                self.DBSession.query(LayerWMS).filter(LayerWMS.ogc_server_id == self.ogcserver.id).all()
+                session.query(LayerWMS)
+                .options(joinedload(LayerWMS.restrictionareas).subqueryload(RestrictionArea.roles))
+                .filter(LayerWMS.ogc_server_id == self.ogcserver.id)
+                .all()
```
Comment (arnaud-morvan) [inline docker/qgisserver/geomapfish_qgisserver/accesscontrol.py:283]: May use a subqueryload for restrictionareas also as we won't make a product between layers and restriction areas.

File: docker/qgisserver/tests/functional/conftest.py [hunk 2]
```diff
@@ -33,7 +33,10 @@ def settings():
 
 @pytest.fixture(scope="module")
 def scoped_session(dbsession):
```
Comment (arnaud-morvan) [inline docker/qgisserver/tests/functional/conftest.py:35]: This fixture might be renamed

**Comment by sbrunner** [comment]
@arnaud-morvan Do you want to review it again?

File: docker/qgisserver/geomapfish_qgisserver/accesscontrol.py [hunk 3]
```diff
@@ -43,13 +43,23 @@
...
 
 
+def create_scoped_session(url, config):
+    configure_mappers()
+    db_match = re.match(".*(@[^@]+)$", url)
+    LOG.info("Connect to the database: ***%s", db_match.group(1) if db_match else "")
+    engine = sqlalchemy.create_engine(url, **config)
+    session_factory = sessionmaker()
+    session_factory.configure(bind=engine)
+    return scoped_session(session_factory)
```
[the remark this answers is not in the document]
Comment (sbrunner) [reply docker/qgisserver/geomapfish_qgisserver/accesscontrol.py:60]: Effectively it looks correct, I can try to remove it...

**Comment by sbrunner** [comment]
@arnaud-morvan Updated, now I think that's OK :-)

File: docker/qgisserver/tests/functional/accesscontrol_allow_to_edit_test.py [hunk 4]
```diff
@@ -43,37 +43,54 @@
...
+    dbsession.execute(layer_ra.delete())
+    dbsession.query(TreeItem).delete()
+    dbsession.query(OGCServer).delete()
+    dbsession.execute(role_ra.delete())
+    dbsession.query(RestrictionArea).delete()
+    dbsession.execute(user_role.delete())
+    dbsession.query(User).delete()
+    dbsession.query(Role).delete()
+    dbsession.commit()
+    dbsession.close()
```
Comment (arnaud-morvan) [inline docker/qgisserver/tests/functional/accesscontrol_allow_to_edit_test.py:70]: This session cleaning seems duplicated many times, may be factorized.

**Pull request merged**